File: golang/day01.go
package golang

import (
	"fmt"
	"strconv"
	"strings"
)

func Day01(rawInput string) {
	fmt.Print("This is day 01 code:\n\n")

	var inputs []int

	for _, value := range strings.Split(rawInput, "\n") {
		i, _ := strconv.Atoi(value)
		if i > 0 {
			inputs = append(inputs, i)
		}
	}

	var partOne, partTwo int

	for _, firstNum := range inputs {
		for _, secondNum := range inputs {
			if firstNum+secondNum == 2020 {
				//fmt.Printf("Part one solution: %v (numbers: %v, %v)\n", firstNum*secondNum, firstNum, secondNum)
				if partOne == 0 {
					partOne = firstNum * secondNum
				}
			}
			for _, thirdNum := range inputs {
				if firstNum+secondNum+thirdNum == 2020 {
					//fmt.Printf("Part two solution: %v (numbers: %v, %v and %v)\n", firstNum*secondNum*thirdNum, firstNum, secondNum, thirdNum)
					if partTwo == 0 {
						partTwo = firstNum * secondNum * thirdNum
					}
				}
			}
		}
	}

	fmt.Printf("Part one solution: %v\n", partOne)
	fmt.Printf("Part two solution: %v\n", partTwo)
	// took 33 min
	// executed in 7.966ms (Ryzen 5 3600)
}

File: 2020/aocjava/Day01.java
package aocjava;

import java.util.List;

class Day01 {
    public static void main(String[] args) {
        System.out.println("\nAdvent of Code day 1:");

        List<Integer> input = Utils.promptInputNumbers();

        int partOneSolution = 0;
        int partTwoSolution = 0;

        for (int firstNum : input) {
            for (int secondNum : input) {
                if(firstNum+secondNum == 2020){
                    if(partOneSolution == 0){
                        partOneSolution = firstNum*secondNum;
                    }
                    else if(partOneSolution != firstNum*secondNum){
                        System.out.println("There might be more solutions to part one!");
                    }
                }
                for (int thirdNum : input) {
                    if(firstNum+secondNum+thirdNum == 2020){
                        if(partTwoSolution == 0){
                            partTwoSolution = firstNum*secondNum*thirdNum;
                        }
                        else if(partTwoSolution != firstNum*secondNum*thirdNum){
                            System.out.println("There might be more solutions to part two!");
                        }
                    }
                }
            }
        }

        System.out.println("\nPart one solution: "+partOneSolution);
        System.out.println("Part two solution: "+partTwoSolution);

    }
}